base/int.jl

```diff
@@ -753,6 +757,12 @@ julia> big"7891.5"
 julia> big"_"
 ERROR: ArgumentError: invalid number format _ for BigInt or BigFloat
 [...]
+
+julia> foo() = big"1"
+foo (generic function with 1 method)
+
+julia> x, y = foo(), foo() # `x` and `y` point to the same BigInt instance
```

Comment:
If we want to document this guarantee, this should also check for `===` instead of just mentioning it in a comment.